ProcessOptimizer/plots.py

```diff
@@ -1555,6 +1556,228 @@ def plot_objective_1d(
+    is_cat = [isinstance(dim, Categorical) for dim in space.dimensions]
+    # Check if we have any integer dimensions, as this influences the plots
+    is_int = [isinstance(dim, Integer) for dim in space.dimensions]
+    # Identify the location of the expected minimum, and its mean and std
+    x_eval, [res_mean, res_std] = expected_minimum(
+        result,
+        n_random_starts=20,
+        random_state=None,
+        return_std=True,
+    )
+        
+    rvs_transformed = space.transform(space.rvs(n_samples=n_samples))
+    _, minimum, _ = _map_categories(space, result.x_iters, x_eval)
+    
+    # Gather all data relevant for plotting
+    plots_data = []
+    # Starting point for our maximum visible quality
+    plot_max_q = max_quality
+    
+    for i in range(space.n_dims):
+        row = []
+        xi, yi, stddevs = dependence(
+            space,
+            result.models[-1],
+            i,
+            j=None,
+            sample_points=rvs_transformed,
+            n_points=n_points,
+            x_eval=x_eval,
+        )
+        row.append({"xi": xi, "yi": yi, "std": stddevs})
+        plots_data.append(row)
+        # Check if our quality interval goes above our desired max_quality view
+        if -np.min(yi-1.96*stddevs) > plot_max_q:
+            plot_max_q = -np.min(yi-1.96*stddevs)
+    
+    # Create the list to store figure handles
+    figure_list = []  
+    
+    # Expand maximal quality value if needed to show upper bound on uncertainty
+    max_quality = plot_max_q
+    
+    # Build all the plots in the figure
+    for n in range(space.n_dims):
+        # Prepare a figure    
+        fig, ax_ = plt.subplots(
+            figsize=(size, size),
+            dpi=200,
+        )
+        # Set the padding
+        fig.subplots_adjust(
+            left=0.12, right=0.93, bottom=0.2, top=0.95, hspace=0.0, wspace=0.0
+        )
+
+        # Get data to plot in this subplot
+        xi = plots_data[n][0]["xi"]
+        yi = plots_data[n][0]["yi"]
+        stddevs = plots_data[n][0]["std"]        
+        
+        # Set y-axis limits with a small buffer
+        ax_.set_ylim(0, max_quality*1.02)
+        
+        # Enter here when we plot a categoric factor
+        if is_cat[n]:
+            # Expand the x-axis for this factor so we can see the first
+            # and the last category
+            ax_.set_xlim(np.min(xi)-0.2, np.max(xi)+0.2)            
+            
+            # Create one uniformly colored bar for each category.
+            # Edgecolor ensures we can see the bar when plotting 
+            # at best obeservation, as stddev is often tiny there
+            ax_.bar(
+                xi,
+                2*1.96*stddevs,
+                width=0.2,
+                bottom=(-yi-1.96*stddevs),
+                alpha=0.5,
+                color="green",
+                edgecolor="green",
+            )
+            # Highlight the expected minimum
+            ax_.scatter(

```

Comment:
In the continuous examples, we don't show the expected y value, and instead have the expected minimum as a vertical dotted line. Is it on purpose that this is different?

Reply:
After thinking on it, interpretation is probably easier when it is consistent across all factors, so I have reverted this to a vertical dashed line across all plots. Incidentally this makes the code a lot simpler, which is a plus.